File: assertion-roulette/python-unittest/GlobalQueryFiltersTests_Should_Add_Modification_Auditing_Properties.py
import unittest

class TestGlobalQueryFilters(unittest.TestCase):
    def test_should_add_modification_auditing_properties(self):
        result = default_test_db_context.products.add(Product(name="Product Name", code="product_code"))
        default_test_db_context.save_changes()
        
        default_test_db_context.products.update(result.entity)
        default_test_db_context.save_changes()
        
        db_context_to_get_updated_entity = get_default_test_db_context()
        updated_entity = db_context_to_get_updated_entity.products.find(result.entity.id)
        
        self.assertIsNotNone(updated_entity.modifier_user_id)
        self.assertNotEqual("", updated_entity.modifier_user_id)
        self.assertIsNotNone(updated_entity.modification_time)
        self.assertEqual(datetime.today().strftime("%Y-%m-%d"), updated_entity.modification_time.strftime("%Y-%m-%d"))

File: assertion-roulette/csharp-xunit/GlobalQueryFiltersTests_Should_Add_Modification_Auditing_Properties.cs
using Xunit;

namespace BackOfficeBase.Tests.DataAccess
{
    public class GlobalQueryFiltersTests
    {
        [Fact]
        public void Should_Add_Modification_Auditing_Properties()
        {
            {
                var result = await DefaultTestDbContext.Products.AddAsync(new Product { Name = "Product Name", Code = "product_code" });
                await DefaultTestDbContext.SaveChangesAsync();
                DefaultTestDbContext.Products.Update(result.Entity);
                await DefaultTestDbContext.SaveChangesAsync();
                var dbContextToGetUpdatedEntity = GetDefaultTestDbContext();
                var updatedEntity = await dbContextToGetUpdatedEntity.Products.FindAsync(result.Entity.Id);
                Assert.NotNull(updatedEntity.ModifierUserId);
                Assert.NotEqual(Guid.Empty, updatedEntity.ModifierUserId);
                Assert.NotNull(updatedEntity.ModificationTime);
                Assert.Equal(DateTime.Today.ToShortDateString(), updatedEntity.ModificationTime.Value.ToShortDateString());
            }
        }
    }
}
